nav2_core/include/nav2_core/navigator.hpp

```diff
@@ -198,7 +294,7 @@ class Navigator
    * @brief Deactivation of the navigator's backend BT and actions
    * @return bool If successful
    */
-  bool on_deactivate()
+  bool on_deactivate() override
```

Comment:
This is kind of a mess, why are we not calling configure / activate / deactivate / cleanup overrides that we established where the `NavigatorBase` does the `on_XYZ` variants? That's super confusing that there are both `configure` and `on_configure` and now 2 of the `on_` variants rather than using the `configure` set up and currently empty

Reply:
E.g. `NavigatorBase` should have all the `on_XYZ` functions, then the `Navigator` should be implementing just the `XYZ` functions that are called as part of `on_XYZ` in the base class to preform work.